extensions/2.0/Khronos/KHR_mesh_instancing/README.md

```diff
@@ -0,0 +1,64 @@
+# KHR\_mesh\_instancing
+
+## Khronos 3D Formats Working Group
+
+* John Cooke
+* Don McCurdy
+* Arseny Kapoulkine 
+
+## Acknowledgments
+
+* TODO
+
+## Status
+
+Experimental
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension is specfically designed to enable GPU instancing which renders many copies of a single mesh at once using a small number of draw calls.  It's useful for things 
+like trees, grass, road signs, etc.  The TRANSLATION, ROTATION, and SCALE attributes allows the mesh to be displayed at many different locations with different rotations and scales.  
+Custom attributes can use the underscore mechanism to achieve other effects (i.e. _ID, _TRANSFORM4x3, etc.).
+
+## Extending Nodes with per instance attributes
+
+Instancing is defined by adding the `KHR_mesh_instancing` extension to any glTF node that has a mesh.  Instancing only applies to mesh nodes, there is no defined behavior for a node 
+with this extension that doesn't also have a mesh.  Applying to nodes rather than meshes allows the same mesh to be used by several nodes, instanced or otherwise.  The attributes 
+section contains accessor ids for the TRANSLATION, ROTATION, and SCALE attribute buffers, all of which are optional.  The attributes specify an object space transform that should be 
+multipled by the node's world transform in the shader to produce the final world transform for the instance. For example, the following defines some instancing attributes to a node with mesh.  
+
+```json
+{
+    "nodes": [
+        {
+            "mesh": 0,
+            "name": "teapot",
+            "extensions": {
+                "KHR_mesh_instancing": {
+                    "attributes": {
+                        "TRANSLATION": 0,
+                        "ROTATION": 1,
+                        "SCALE": 2,
+                    },
```

Comment:
Is there a reason we do not specify the instance count in the extension? I can foresee this causing security vulnerabilities in native clients due to malformed/mismatched accessor counts for the different attributes.

Reply:
This follows existing convention from the core spec: `mesh.primitive.attributes` relies on `accessor.count` being the same for all attributes. Having a dedicated property to signal the instance count does not add much value here as implementations would still need to check that it's less than or equal to each `accessor.count`.

Reply:
Makes sense, it wouldn't be wise to break from the convention here.

Reply:
A possible counterargument is that with the core `mesh` definition, the `POSITION` attribute is required. Here, none of the attributes are required, and implementations might need even need to look to the custom attributes to find the actual instance count. In the three.js implementation this becomes:

```js
var template = translation || rotation || scale || otherAccessors[ 0 ];
var count = template.count;
```

That's a little ugly, but not burdensome in our case. But if other implementations think it's useful to know the count more directly I wouldn't be opposed to putting it in the extension.